test-case/check-runtime-pm-boundary.sh

```diff
@@ -0,0 +1,165 @@
+    dev=$(func_pipeline_parse_value $idx dev)
+    pcm=$(func_pipeline_parse_value $idx pcm)
+
+    for pm_sleep in ${arr_sleep[@]}
+    do
+        for i in $(seq 1 $loop_count)
+        do
+            dlogi "Iteration [$idx] [$i of $loop_count] for $pcm, [sleep $pm_sleep]"
+            # playback device - check status
+            dlogc "$cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q"
+            $cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q &
+            pid=$!
+
+            # TODO: delay 2.5s is workaround for the SSH aplay delay issue.
+            sleep 2.5
+
+            kill -0 $pid
+            [[ $? -ne 0 ]] && dloge "$cmd process for pcm $pcm is not alive" && exit 1
+
+            [[ -d /proc/$pid ]] && result=`cat $runtime_status`
+
+            dlogi "While $cmd running, runtime status: $result"
+            if [[ $result == active ]]; then
+                # stop playback device - check status again
+                dlogc "kill process: kill -9 $pid"
+                kill -9 $pid && wait $pid 2>/dev/null
+                dlogi "$cmd end"
+
+                dlogc "sleep $pm_sleep"
+                sleep $pm_sleep
+
+                result=`cat $runtime_status`
+                dlogi "After $cmd killed and sleep $pm_sleep, runtime status: $result"
+                if [ $pm_sleep -gt $state_delay ] && [ $result != suspended ]; then
+                    # check runtime pm status ONLY when pm_sleep is greater than expected state transition time
+                    dloge "$cmd process for pcm $pcm runtime status is not suspended as expected"
+                    exit 1
+                fi
+
+                dlogi "check $cmd can start without any problem after sleep $pm_sleep"
+                dlogc "$cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q"
+                $cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q &
+                pid=$!
+
+                sleep 1
+
+                kill -0 $pid
+                [[ $? -ne 0 ]] && dloge "$cmd process for pcm $pcm is not alive" && exit 1
+
+                kill -9 $pid && wait $pid 2>/dev/null
+                dlogi "$cmd end"
+
+            else
+                dloge "$cmd process for pcm $pcm runtime status is not active as expected"
+                # stop playback device otherwise no one will stop this $cmd.
+                dlogc "kill process: kill -9 $pid"
+                kill -9 $pid && wait $pid 2>/dev/null
+                exit 1
+            fi
+
+        done # end of loop_count
+
+        sof-kernel-log-check.sh 0
+    done # end of pm_sleep
+
+    #runtime pm status check at end the end of each pipeline test
+    dlogc "sleep $state_delay at the end"
+    sleep $state_delay
+    result=`cat $runtime_status`
+    dlogi "runtime status: $result"
+    if [[ $result != suspended ]]; then
+        dloge "$cmd process for pcm $pcm runtime status is not suspended as expected"
+        exit 1
+    fi
```

Comment:
the original case main logic for the loop
```
loop begin
$cmd # to run cmd
check $runtime_status
is active
    stop $cmd
    check $runtime_status
    is suspended
loop end
```
your new case logic:
```
loop begin
$cmd # to run cmd
check $runtime_status
is active
    stop $cmd
    check $runtime_status
    is suspended
    $cmd # to run cmd
loop end
is suspended
```

So I couldn't see the different between the original 
Because those check will cover by the loop
so if you want check `suspended` you can add at original loop begin  

The different between this case with original, I think this case will catch at `$i`, the original will catch at `$i + 1`

Reply:
@Bin-QA the main purpose of this test is check aplay/arecord again just before/after runtime pm, having said that probably sleep [3 5 7] are most important delay time to start playback or capture again. There was known issue, it didn't work before.

Adding this logic to known runtime pm test case, it looks to me messy.

Reply:
how about give the random time range and increase the loop